dm_control/locomotion/walkers/assets/dog_v2/build_dog.py

```diff
@@ -241,186 +298,260 @@ def main(argv):
+      dmp = physics.bind(j).damping[0]
+      stf = physics.bind(joint_acts[i]).gainprm[0]
+      arma = physics.bind(j).armature[0]
+      print(
+          "{:20} {:10.4} {:10} {:10.4} {:10.4} {:10}".format(
+              j.name,
+              mass_eff,
+              dmp,
+              stf,
+              dmp / (2 * np.sqrt(mass_eff * stf)),
+              arma,
+          )
+      )
 
-  print('Finalising and saving model.')
-  xml_string = model.to_xml_string('float', precision=4, zero_threshold=1e-7)
+  if composer:
+    torso = model.find("body", "torso")
+    torso.pos = np.zeros(3)
+    
+  print("Finalising and saving model.")
+  xml_string = model.to_xml_string("float", precision=4, zero_threshold=1e-7)
   root = etree.XML(xml_string, etree.XMLParser(remove_blank_text=True))
 
-  print('Remove hashes from filenames')
-  assets = list(root.find('asset').iter())
+  print("Remove hashes from filenames")
+  assets = list(root.find("asset").iter())
   for asset in assets:
-    asset_filename = asset.get('file')
+    asset_filename = asset.get("file")
     if asset_filename is not None:
       name = asset_filename[:-4]
       extension = asset_filename[-4:]
-      asset.set('file', name[:-41] + extension)
+      if asset_filename[-3:] == "skn":
+        asset.set("file", "skins/" + name[:-41] + extension)
+      else:
+        asset.set("file", name[:-41] + extension)
 
-  print('Add <compiler meshdir/>, for locally-loadable model')
+  print("Add <compiler meshdir/>, for locally-loadable model")
   compiler = etree.Element(
-      'compiler', meshdir=ASSET_RELPATH, texturedir=ASSET_RELPATH
+      "compiler", meshdir=ASSET_RELPATH, texturedir=ASSET_RELPATH
   )
   root.insert(0, compiler)
 
   print('Remove class="/"')
-  default_elem = root.find('default')
+  default_elem = root.find("default")
   root.insert(6, default_elem[0])
   root.remove(default_elem)
   xml_string = etree.tostring(root, pretty_print=True)
-  xml_string = xml_string.replace(b' class="/"', b'')
+  xml_string = xml_string.replace(b' class="/"', b"")
 
-  print('Insert spaces between top level elements')
+  print("Insert spaces between top level elements")
   lines = xml_string.splitlines()
   newlines = []
   for line in lines:
     newlines.append(line)
-    if line.startswith(b'  <'):
-      if line.startswith(b'  </') or line.endswith(b'/>'):
-        newlines.append(b'')
-  newlines.append(b'')
-  xml_string = b'\n'.join(newlines)
+    if line.startswith(b"  <"):
+      if line.startswith(b"  </") or line.endswith(b"/>"):
+        newlines.append(b"")
+  newlines.append(b"")
+  xml_string = b"\n".join(newlines)
 
+  name_prefix = "composer_" if composer else ""
   # Save to file.
-  f = open('dog.xml', 'wb')
+  if not use_muscles:
+    name = name_prefix + "dog.xml"
+  else:
+    name = name_prefix + "dog_muscles_{}_{}.xml".format(
+        muscle_strength_scale, muscle_dynamics)
+
+  f = open(os.path.join(CURRENT_DIR, name), "wb")
```

Comment:
Is this script meant to be run by anyone but dm_control developers?

Seems weird to be writing into the source directory - I would expect the output directory to be another flag argument to the binary.